File: tasks/translation/output/Qwen/Qwen2.5-Coder-32B-Instruct-AWQ/codenet/direct/comment/uncomment/direct/Java/C++/s824230885_commented.cpp
#include <iostream>
#include <vector>
#include <string>
#include <sstream>

using namespace std;

class InputIterator {
    vector<string> inputLine;
    int index = 0;
    int max;
    string read;
public:
    InputIterator() {
        string line;
        while (getline(cin, line)) {
            inputLine.push_back(line);
        }
        max = inputLine.size();
    }

    bool hasNext() { return (index < max); }

    string next() {
        if (hasNext()) {
            return inputLine[index++];
        } else {
            throw out_of_range("There is no more input");
        }
    }
};

InputIterator ii; // This class cannot be used in reactive problem.
void flush() { cout.flush(); }
void myout(const auto& t) { cout << t << endl; }
void myerr(const auto& t) { cerr << "debug:" << t << endl; }
string next() { return ii.next(); }
bool hasNext() { return ii.hasNext(); }
int nextInt() { return stoi(next()); }
long long nextLong() { return stoll(next()); }
double nextDouble() { return stod(next()); }

vector<string> nextStrArray() { return myconv(next(), 8); }
vector<string> nextCharArray() { return myconv(next(), 0); }

vector<int> nextIntArray() {
    vector<string> input = nextStrArray();
    vector<int> ret(input.size());
    for (int i = 0; i < input.size(); i++) {
        ret[i] = stoi(input[i]);
    }
    return ret;
}

vector<long long> nextLongArray() {
    vector<string> input = nextStrArray();
    vector<long long> ret(input.size());
    for (int i = 0; i < input.size(); i++) {
        ret[i] = stoll(input[i]);
    }
    return ret;
}

string myconv(const auto& list, int no) { // only join
    string joinString;
    if (no == 8) joinString = " ";
    else if (no == 0) joinString = "";
    else throw invalid_argument("Invalid conversion type");

    if constexpr (is_same_v<decltype(list), vector<string>>) {
        return join(list, joinString);
    } else {
        throw invalid_argument("Don't join");
    }
}

vector<string> myconv(const string& str, int no) { // only split
    string splitString;
    if (no == 8) splitString = " ";
    else if (no == 0) splitString = "";
    else throw invalid_argument("Invalid conversion type");

    vector<string> result;
    stringstream ss(str);
    string item;
    while (getline(ss, item, splitString[0])) {
        result.push_back(item);
    }
    return result;
}

string join(const vector<string>& v, const string& delimiter) {
    string result;
    for (size_t i = 0; i < v.size(); ++i) {
        result += v[i];
        if (i != v.size() - 1) {
            result += delimiter;
        }
    }
    return result;
}

void solve() { // Here is the main function
    int N = nextInt();
    vector<int> list = nextIntArray();
    int max = 0;
    int output = 0;
    for (int i = 0; i < N; i++) {
        if (max <= list[i]) {
            max = list[i];
            output++;
        }
    }
    myout(output);
}

int main() {
    solve();
    flush();
    return 0;
}
// End of Code

File: tasks/translation/output/Qwen/Qwen2.5-32B-Instruct-AWQ/codenet/direct/comment/comment/direct/Java/Python/s824230885_commented.py
import sys

class InputIterator:
    def __init__(self):
        self.input_line = []
        for line in sys.stdin:
            self.input_line.append(line.strip())
        self.index = 0
        self.max = len(self.input_line)

    def hasNext(self):
        return self.index < self.max

    def next(self):
        if self.hasNext():
            return self.input_line[self.index++]
        else:
            raise IndexError("There is no more input")

ii = InputIterator()
out = sys.stdout.write

def flush():
    sys.stdout.flush()

def myout(t):
    out(f"{t}\n")
    flush()

def myerr(t):
    sys.stderr.write(f"debug: {t}\n")

def next():
    return ii.next()

def hasNext():
    return ii.hasNext()

def nextInt():
    return int(next())

def nextLong():
    return int(next())

def nextDouble():
    return float(next())

def nextStrArray():
    return next().split(" ")

def nextCharArray():
    return list(next())

def nextIntArray():
    return [int(x) for x in nextStrArray()]

def nextLongArray():
    return [int(x) for x in nextStrArray()]

def myconv(lst, no):  # only join
    join_string = " " if no == 8 else "\n" if no == 9 else ""
    if isinstance(lst, list):
        return join_string.join(map(str, lst))
    else:
        raise TypeError("Don't join")

def solve():
    N = nextInt()
    lst = nextIntArray()
    max_val = 0
    output = 0
    for i in range(N):
        if max_val <= lst[i]:
            max_val = lst[i]
            output += 1
    myout(output)

if __name__ == "__main__":
    solve()  # End of Code
